### PR TITLE
Add merge transition/finalization banners

### PR DESCRIPTION
**Pull request by g11tech**
**Motivation**
This PR adds a merge transition/merge finalization banner capabilities plus additional helpful log around merge block/finalization.

Also fixes the flag of `isMergeTransitionBlock` on the flags passed to the forkchoice. Earlier it was being evaluated on a `postState` which was problematic, because even on the merge transition block, postState would already have show the mergeCompleted. Fixed it to evaluate on `preState`.

PS: Banners could be updated later as per contributions to https://github.com/ChainSafe/lodestar/issues/3961 (which is a separate issue of having the banner art contributions, hence this PR doesn't close that issue, but only enables it)

### CODE REVIEW

File: packages/lodestar/src/chain/banners/pandaFinalized.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-useless-escape */
+export const POS_PANDA_FINALIZED_BANNER = `
```
Comment (g11tech) [inline packages/lodestar/src/chain/banners/pandaFinalized.ts]: @philknows , do we want a similar banner on the finalization of the `merge`, i.e. first finalized checkpoint post merge? we can remove this or make it super minimal or have something else about it.

Comment (philknows) [reply packages/lodestar/src/chain/banners/pandaFinalized.ts]: I think just the merge transition block is enough for the banner. It's what the other clients are doing. Finalization can just remain a normal log statement.

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 2]
```diff
@@ -118,12 +123,37 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
...
+    chain.logger.info("Execution transitioning from PoW to PoS!!!");
+    chain.logger.info("Imported terminal PoW block", {
+      blockHash: mergeBlockHash,
+      executionHash: mergeExecutionHash,
+      powHash: mergePowHash,
+    });
+  }
+  if (!prevIsMergeTransitionFinalized && chain.forkChoice.getMergeTransitionFinalized()) {
+    const finalizedCheckpoint = chain.forkChoice.getFinalizedCheckpoint();
+    chain.logger.info(POS_PANDA_FINALIZED_BANNER);
```
Comment (g11tech) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: @philknows yea this banner can go, or it can be super minimal, or we can add it later now we have a spot

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 3]
```diff
@@ -118,12 +123,36 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
...
+    const mergeBlockHash = toHexString(ssz.bellatrix.BeaconBlock.hashTreeRoot(mergeBlock));
+    const mergeExecutionHash = toHexString(mergeBlock.body.executionPayload.blockHash);
+    const mergePowHash = toHexString(mergeBlock.body.executionPayload.parentHash);
+    chain.logger.info(POS_PANDA_TRANSITIONING_BANNER);
+    chain.logger.info("Execution transitioning from PoW to PoS!!!");
+    chain.logger.info("Imported block referencing terminal PoW block", {
+      blockHash: mergeBlockHash,
+      executionHash: mergeExecutionHash,
+      powHash: mergePowHash,
+    });
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: To keep this function cleaner we can move this logic into a function such as
```ts
   if (onBlockPrecachedData.isMergeTransitionBlock) {
     logOnPowBlock(chain, block)
   } else if (!chain.forkChoice.getMergeTransitionFinalized() && chain.forkChoice.getMergeTransitionFinalized()) {
     logOnPowBlockFinalized(chain, block)
   }
```

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 4]
```diff
@@ -118,12 +123,36 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
...
   const prevFinalizedEpoch = chain.forkChoice.getFinalizedCheckpoint().epoch;
+  const prevIsMergeTransitionFinalized = chain.forkChoice.getMergeTransitionFinalized();
+
   chain.forkChoice.onBlock(block.message, postState, onBlockPrecachedData);
 
+  // If this was a merge transition block, it has been successfully imported
+  // in forkchoice! Its now PoS all the way!!!!
+  if (onBlockPrecachedData.isMergeTransitionBlock) {
+    const mergeBlock = block.message as bellatrix.BeaconBlock;
+    const mergeBlockHash = toHexString(ssz.bellatrix.BeaconBlock.hashTreeRoot(mergeBlock));
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: This is not safe, you must use config to get the block type from the slot. There will be more forks after bellatrix

File: packages/lodestar/src/chain/banners/pandaTransitioning.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
+/* eslint-disable no-useless-escape */
+export const POS_PANDA_TRANSITIONING_BANNER = `
```
Comment (dapplion) [inline packages/lodestar/src/chain/banners/pandaTransitioning.ts]: Just put this in utils, not sure we will have that many banners hahaha

File: packages/fork-choice/src/forkChoice/forkChoice.ts [hunk 6]
```diff
@@ -362,6 +363,17 @@ export class ForkChoice implements IForkChoice {
...
       this.fcStore.finalizedCheckpoint = toCheckpointWithHex(finalizedCheckpoint);
+      if (
+        !this.isMergeTransitionFinalized &&
+        bellatrix.isBellatrixStateType(state) &&
+        bellatrix.isBellatrixBlockBodyType(block.body) &&
+        bellatrix.isExecutionEnabled(state, block.body)
+      ) {
+        const finalizedBlock = this.getFinalizedBlock();
+        if (finalizedBlock.executionStatus !== ExecutionStatus.PreMerge) {
+          this.isMergeTransitionFinalized = true;
```
Comment (dapplion) [inline packages/fork-choice/src/forkChoice/forkChoice.ts]: Is this strategy sufficient to not log again after restarting?

Comment (dapplion) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: Moving this logic here just for the log doesn't feel great. Is lighthouse doing the same?

Comment (g11tech) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: lighthouse doesn't seem to be logging finalized, i wanted to add this on my own. 
on restart, if if it starts post merge, isMergeTransitionFinalized should be set true on init and shouldn't log.

Comment (g11tech) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: over here: https://github.com/ChainSafe/lodestar/pull/3963/files#diff-a941ba900b315f252132684ebd4701cf9b4efa664c99112856e3d0313a4ceee5R79

Comment (dapplion) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: While I know it's not a lot, I'm not super keen on adding _some_ complexity for a non critical item like this. While we (coredevs) celebrate finalization users may only care about the merge block, and it's very easy to detect once. Let me know what you think.

Comment (philknows) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: I also have no solicited any banner work for finalized at this time.

Comment (g11tech) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: ok no issues, removed :slightly_smiling_face:

File: packages/lodestar/src/chain/blocks/verifyBlock.ts [hunk 7]
```diff
@@ -290,5 +288,10 @@ export async function verifyBlockStateTransition(
     });
   }
 
-  return {postState, executionStatus};
+  const isMergeTransitionBlock =
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/verifyBlock.ts]: Why has this variable been moved to verifyBlock?

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 8]
```diff
@@ -118,12 +123,17 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
...
     onBlockPrecachedData.powBlock = powBlock;
     onBlockPrecachedData.powBlockParent = powBlockParent;
-    onBlockPrecachedData.isMergeTransitionBlock = true;
   }
 
   const prevFinalizedEpoch = chain.forkChoice.getFinalizedCheckpoint().epoch;
   chain.forkChoice.onBlock(block.message, postState, onBlockPrecachedData);
 
+  // If this was a merge transition block, it has been successfully imported
+  // in forkchoice! Its now PoS all the way!!!!
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: For consistency with styling guides of prettier, comments should target close and less than 120 characters in width

File: packages/lodestar/src/chain/blocks/verifyBlock.ts (hunk 7 above)
Comment (g11tech) [reply packages/lodestar/src/chain/blocks/verifyBlock.ts]: need preState to evaluate it, on postState it was always coming as false as on merge block poststate will have merge complete

**Comment by dapplion** [comment]
Note this PR was blocked by https://github.com/ChainSafe/lodestar/pull/3984

https://github.com/ChainSafe/lodestar/pull/3984 is now merged :heavy_check_mark:

File: packages/lodestar/src/chain/blocks/verifyBlock.ts [hunk 9]
```diff
@@ -317,5 +316,24 @@ export async function verifyBlockStateTransition(
...
 
+  // All checks have passed, if this is a merge transition block we can log
+  if (isMergeTransitionBlock) {
+    logOnPowBlock(chain, block);
+  }
+
   return {postState, executionStatus};
 }
+
+function logOnPowBlock(chain: VerifyBlockModules, block: allForks.SignedBeaconBlock): void {
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/verifyBlock.ts]: I think it would be safer to have
```ts
function logOnPowBlock(chain: VerifyBlockModules, block: bellatrix.BeaconBlock): void
```
and do the casting right below `isMergeTransitionBlock` where it's clear that the casting is safe

File: packages/lodestar/src/chain/blocks/verifyBlock.ts [hunk 10]
```diff
@@ -43,14 +44,13 @@ export async function verifyBlock(
 ): Promise<FullyVerifiedBlock> {
   const parentBlock = verifyBlockSanityChecks(chain, partiallyVerifiedBlock);
 
-  const {postState, executionStatus} = await verifyBlockStateTransition(chain, partiallyVerifiedBlock, opts);
+  const processedBlockData = await verifyBlockStateTransition(chain, partiallyVerifiedBlock, opts);
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/verifyBlock.ts]: Why this change? Being explicit and not doing the spread feels more readable in my opinion

**Review by dapplion — APPROVED** [review]
